app/vite.config.js

```diff
@@ -62,7 +62,6 @@ export default defineConfig({
 			proxy: {
 				'^/(?!admin)': {
 					target: process.env.API_URL ? process.env.API_URL : 'http://127.0.0.1:8055/',
-					changeOrigin: true,
```

Comment:
As the origin is expected to be from the API is this necessary anymore?

Reply:
The redirect param can be a relative path (ex: /admin), not always a full URL. 
In multi-domain setups with a relative redirect, we need to know which frontend initiated the request to redirect back correctly. 
That's why we capture the request's host, it's the only way to determine the origin when the redirect is relative

Reply:
We also set the cookie on the [response](https://github.com/directus/directus/blob/2f41951857a540ba2ab5e5e829d29f0a9c9661ae/api/src/auth/drivers/oauth2.ts#L496) irrespective of the originating host

Reply:
Yes, res.cookie() sets the cookie on the current request's domain. That's exactly why the callback URL must match the originating host, so the auth cookie is set on the domain where the user will be browsing. 
That's why we should remove `changeOrigin` now 😅

Reply:
The callback url will not match the originating host as it will always be the API url (e.g. `http://www.example.com/auth/login/keycloak/callback`) even if the originating request came from a different domain (e.g. `www.myapp.com`). The cookie will be set on whatever domain is specified in the env cookie settings.

As this is a dev only change, it should not effect anything removing it 👍 .